File: ks/read_pdf.py
import PyPDF2
import sys

def main():
    pdf_path = sys.argv[1]
    out_path = sys.argv[2]
    try:
        reader = PyPDF2.PdfReader(pdf_path)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        with open(out_path, 'w', encoding='utf-8', errors='ignore') as f:
            f.write(text)
        print("Success")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: ks/read_pdf.js
import { PdfReader } from "pdfreader";
import fs from "fs";

const filePath = process.argv[2];
const reader = new PdfReader();

let text = "";

reader.parseFileItems(filePath, (err, item) => {
  if (err) {
    console.error("error:", err);
  } else if (!item) {
    fs.writeFileSync("ks/temp_pdf.txt", text, "utf8");
    console.log("Done extracting.");
  } else if (item.text) {
    text += item.text + " ";
  }
});
